### PR TITLE
Fixing compilation&clippy warnings

### PR DESCRIPTION
**Pull request by tomusdrw**

### CODE REVIEW

File: util/src/trie/sectriedb.rs [hunk 1]
```diff
@@ -50,7 +50,7 @@ impl<'db> SecTrieDB<'db> {
 }
 
 impl<'db> Trie for SecTrieDB<'db> {
-	fn iter<'a>(&'a self) -> Box<Iterator<Item = (Vec<u8>, &[u8])> + 'a> {
+	fn iter<'a>(&'a self) -> Box<Iterator<Item=TrieItem> + 'a> {
```
Comment (gavofyork) [inline util/src/trie/sectriedb.rs]: missing spacing around `=`

File: util/src/trie/triedb.rs [hunk 2]
```diff
@@ -337,7 +337,7 @@ impl<'db> TrieDB<'db> {
 }
 
 impl<'db> Trie for TrieDB<'db> {
-	fn iter<'a>(&'a self) -> Box<Iterator<Item = (Vec<u8>, &[u8])> + 'a> {
+	fn iter<'a>(&'a self) -> Box<Iterator<Item=TrieItem> + 'a> {
```
Comment (gavofyork) [inline util/src/trie/triedb.rs]: same as above.

File: util/src/trie/trietraits.rs [hunk 3]
```diff
@@ -32,7 +35,7 @@ pub trait Trie {
 	fn get<'a, 'key>(&'a self, key: &'key [u8]) -> Option<&'a [u8]> where 'a: 'key;
 
 	/// Returns an iterator over elements of trie.
-	fn iter<'a>(&'a self) -> Box<Iterator<Item = (Vec<u8>, &[u8])> + 'a>;
+	fn iter<'a>(&'a self) -> Box<Iterator<Item=TrieItem> + 'a>;
```
Comment (gavofyork) [inline util/src/trie/trietraits.rs]: same as above.

File: util/src/trie/sectriedb.rs (hunk 1 above)
Comment (rphmeier) [inline util/src/trie/sectriedb.rs]: This is the canonical style used in `std`: https://doc.rust-lang.org/stable/src/core/up/src/libcore/iter.rs.html#2730

File: util/src/trie/trietraits.rs [hunk 4]
```diff
@@ -17,6 +17,9 @@
 use hash::H256;
 use rlp::SHA3_NULL_RLP;
 
+/// Trie-Item type.
+pub type TrieItem<'a> = (Vec<u8>, &'a[u8]);
```
Comment (rphmeier) [inline util/src/trie/trietraits.rs]: missing space here

**Pull request merged**